packages/react-core/src/components/EmptyState/EmptyState.tsx

```diff
@@ -10,37 +11,78 @@ export enum EmptyStateVariant {
   full = 'full'
 }
 
+export type EmptyStateStatus = 'danger' | 'warning' | 'success' | 'info' | 'custom';
+export type EmptyStateVariantType = 'xs' | 'sm' | 'lg' | 'xl' | 'full';
+
 export interface EmptyStateProps extends React.HTMLProps<HTMLDivElement> {
   /** Additional classes added to the empty state */
   className?: string;
   /** Content rendered inside the empty state */
-  children: React.ReactNode;
+  children?: React.ReactNode;
   /** Modifies empty state max-width and sizes of icon, title and body */
-  variant?: 'xs' | 'sm' | 'lg' | 'xl' | 'full';
+  variant?: EmptyStateVariantType;
   /** Cause component to consume the available height of its container */
   isFullHeight?: boolean;
+  /** Status style options added to the empty state icon */
+  status?: EmptyStateStatus;
+  /** Additional class names to apply to the empty state header */
+  headerClassName?: string;
+  /** Additional props passed to the empty state header container */
+  headerProps?: React.HTMLProps<HTMLDivElement>;
+  /** Additional classes added to the title inside empty state header */
+  titleClassName?: string;
+  /** Text of the title inside empty state header, will be wrapped in headingLevel */
+  titleText: React.ReactNode;
+  /** Empty state icon element to be rendered. Can also be a spinner component */
+  icon?: React.ComponentType<any>;
+  /** Additional props passed to the icon element */
+  iconProps?: EmptyStateIconProps;
+  /** The heading level to use, default is h1 */
+  headingLevel?: EmptyStateHeadingLevel;
```

Comment:
It seems that I need to explicit include the `EmptyStateHeader` props here for our props table to work properly. At some point in the future we should be able to remove this and just have `EmptyStateProps` extend `EmptyStateHeaderProps`, that will require a docs framework update though.